tests/cbmc/proofs/s2n_stuffer_free/s2n_stuffer_free_harness.c

```diff
@@ -22,11 +22,36 @@
 
 void s2n_stuffer_free_harness()
 {
+    /* Non-deterministic inputs. */
     struct s2n_stuffer *stuffer = cbmc_allocate_s2n_stuffer();
 
+    /* Assumptions. */
     nondet_s2n_mem_init();
+    __CPROVER_assume(s2n_result_is_ok(s2n_stuffer_validate(stuffer)));
+    const bool old_alloced = stuffer ? stuffer->alloced : false;
+    struct s2n_blob old_blob;
+    old_blob.data = stuffer ? stuffer->blob.data : NULL;
+    old_blob.growable = stuffer ? stuffer->blob.growable : NULL;
 
-    if (s2n_stuffer_free(stuffer) == S2N_SUCCESS && stuffer != NULL) {
+    /* Operation under verification. */
+    int result = s2n_stuffer_free(stuffer);
+    if (result == S2N_SUCCESS && stuffer != NULL) {
         assert_all_zeroes(stuffer, sizeof(*stuffer));
+    } else if (result == S2N_FAILURE && s2n_errno == S2N_ERR_FREE_STATIC_BLOB) {
+        assert(!old_blob.growable);
     }
+
+    /* Cleanup after expected error cases, for memory leak check. */
+    if ((result == S2N_FAILURE && s2n_errno == S2N_ERR_NOT_INITIALIZED) || !old_blob.growable || !old_alloced) {
```

Comment:
@lrstewart why do we need both `growable` and `alloced` for stuffers? Seems redundant.

Reply:
Not all alloced stuffers are growable. "growable" for a stuffer means that it will automatically resize as needed when you write more data to it.